File: 2.0 java/9120/app/src/main/java/com/example/a9120/SecondActivity.java
package com.example.a9120;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.text.Editable;
import android.view.View;
import android.widget.EditText;

public class SecondActivity extends AppCompatActivity {

    EditText editText;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_second);
        editText = findViewById(R.id.editText);

    }

    public void onOpenThirdActivity(View view) {
        Intent thirdActivityIntent = new Intent (this, ThirdActivity.class);
        String editTextMessage = editText.getText().toString();
        thirdActivityIntent.putExtra("toSecondActivity",editTextMessage); //ключ toSecondActivity уникальный


//        int editNumberMessage = editText.getText().charAt(3);
//        int editNumberMessage = Integer.parseInt(editText.getText().toString());
//        thirdActivityIntent.putExtra("3",editNumberMessage);
        startActivity(thirdActivityIntent);
    }
}

File: 1.0 kotlin/109Intent/app/src/main/java/com/example/a109intent/SecondActivity.kt
package com.example.a109intent

import android.content.Context
import android.content.Intent
import android.os.Bundle
import android.os.Message
import android.os.PersistableBundle
import android.util.Log
import androidx.appcompat.app.AppCompatActivity
import kotlinx.android.synthetic.main.activity_second.*

class SecondActivity: AppCompatActivity(R.layout.activity_second){

    override fun onCreate(savedInstanceState: Bundle?, persistentState: PersistableBundle?) {
        super.onCreate(savedInstanceState, persistentState)
        Log.d("LifeCycleTest", "SecondActivity|onCreate${hashCode()}")
        val message = intent.getStringExtra(KEY_MESSAGE)
        messageTextView.text = message
    }

    companion object{
        private const val KEY_MESSAGE = "message key"

        fun getIntent(context : Context, message: String): Intent {
            return Intent (context,SecondActivity::class.java)
                .putExtra(KEY_MESSAGE,message)
        }
    }

    override fun onStart() {
        super.onStart()
        Log.d("LifeCycleTest", "SecondActivity|onStart${hashCode()}")
    }

    override fun onResume() {
        super.onResume()
        Log.d("LifeCycleTest", "SecondActivity|onResume${hashCode()}")
    }

    override fun onPause() {
        super.onPause()
        Log.d("LifeCycleTest", "SecondActivity|onPause${hashCode()}")
    }

    override fun onStop() {
        super.onStop()
        Log.d("LifeCycleTest", "SecondActivity|onStop${hashCode()}")
    }

    override fun onDestroy() {
        super.onDestroy()
        Log.d("LifeCycleTest", "SecondActivity|onDestroy${hashCode()}")
    }

}